### ISSUE TITLE
Failed assertion: line 2732 pos 14: '_semantics != null': is not true.

### ISSUE THREAD

**Issue by bjornbjorn** [flutter/flutter#12705]
## Steps to Reproduce

This is a menu item called "Logout" in my Drawer:

```
onTap: () async {

                        if(viewModel.isLoggedIn && !viewModel.user.isAnonymous)
                        {
                          // user is logged in, but not as an anonymous user (then we can signout)
                          FirebaseAuth.instance.signOut().then((obj) {
                            Navigator.pop(context);
                          }).catchError((error) {
                            print(error);
                          });

 (...)
```
When clicking the menu item I got this:

```
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
The following assertion was thrown while finalizing the widget tree:
This _ModalScope widget cannot be marked as needing to build because the framework is locked.

When the exception was thrown, this was the stack:
════════════════════════════════════════════════════════════════════════════════════════════════════
```
So I uncommented `// Navigator.of(context).pop()` and hit hot reload to test and then I got the error below (asking me to file this as an issue here).
                        
## Logs

```
══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════
'package:flutter/src/rendering/object.dart': Failed assertion: line 2732 pos 14: '_semantics !=
null': is not true.

Either the assertion indicates an error in the framework itself, or we should provide substantially
more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
https://github.com/flutter/flutter/issues/new

When the exception was thrown, this was the stack:

The following RenderObject was being processed when the exception was fired:
RenderSemanticsAnnotations#95332 NEEDS-PAINT
creator: Semantics ← FocusScope ← _ModalScope-[LabeledGlobalKey<_ModalScopeState>#e4348] ←
_OverlayEntry-[LabeledGlobalKey<_OverlayEntryState>#019cf] ← TickerMode ← _Theatre ←
Overlay-[LabeledGlobalKey<OverlayState>#e25cf] ← _FocusScopeMarker ← Semantics ← FocusScope ←
AbsorbPointer ← Listener ← ⋯
semantics node: SemanticsNode#82
semantic boundary
RenderOffstage#03c40 NEEDS-PAINT
RenderIgnorePointer#1e5a0 NEEDS-PAINT
RenderFractionalTranslation#e1705 NEEDS-PAINT
RenderFractionalTranslation#f852d
RenderDecoratedBox#606b5
════════════════════════════════════════════════════════════════════════════════════════════════════
```

Run `flutter analyze` and attach any output of that command also.

> flutter analyze
Analyzing /Users/bjorn/Documents/workspace/offlinepal...
No issues found!
Ran in 5.7s

## Flutter Doctor

[✓] Flutter (on Mac OS X 10.12.6 16G29, locale en-NO, channel alpha)
    • Flutter at /Users/bjorn/bin/flutter
    • Framework revision e8aa40eddd (7 days ago), 2017-10-17 15:42:40 -0700
    • Engine revision 7c4142808c
    • Tools Dart version 1.25.0-dev.11.0

[✓] Android toolchain - develop for Android devices (Android SDK 26.0.1)
    • Android SDK at /Users/bjorn/Library/Android/sdk
    • Platform android-26, build-tools 26.0.1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] iOS toolchain - develop for iOS devices (Xcode 9.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.0.1, Build version 9A1004
    • ios-deploy 1.9.2
    • CocoaPods version 1.3.1

[✓] Android Studio (version 2.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] IntelliJ IDEA Community Edition (version 2017.2.5)
    • Flutter plugin version 18.1
    • Dart plugin version 172.4155.35

[✓] IntelliJ IDEA Ultimate Edition (version 2017.2.4)
    • Flutter plugin version 17.0
    • Dart plugin version 172.4155.35

[✓] Connected devices
    • iPhone 5s • 75571AD4-E453-4C93-B8AC-689084773DE8 • ios • iOS 11.0 (simulator)
> For more information about diagnosing and reporting Flutter bugs, please see [https://flutter.io/bug-reports/](https://flutter.io/bug-reports/).

**Comment by jonahwilliams** [comment]
Removing hot reload label since the hot reload seems incidental to the crash

**Comment by goderbauer** [comment]
Closing since it is very old and doesn't have a full sample code that reproduces the problem.
